File: src/main/java/sk/hudak/prco/dao/db/GroupProductKeywordsDao.java
package sk.hudak.prco.dao.db;

import sk.hudak.prco.dao.BaseDao;
import sk.hudak.prco.model.GroupProductKeywordsEntity;

import java.util.List;

public interface GroupProductKeywordsDao extends BaseDao<GroupProductKeywordsEntity> {

    GroupProductKeywordsEntity findByGroupId(Long groupId);

    List<String[]> findKeywordsForGroupId(Long groupId);
}

File: src/main/java/sk/hudak/prco/dao/db/GroupProductKeywordsDao.kt
package sk.hudak.prco.dao.db

import sk.hudak.prco.dao.BaseDao
import sk.hudak.prco.model.GroupEntity
import sk.hudak.prco.model.GroupProductKeywordsEntity

interface GroupProductKeywordsDao : BaseDao<GroupProductKeywordsEntity> {

    fun existGroupWithKeywords(groupId: Long, keywords: String): Boolean

    fun findByGroupId(groupId: Long?): List<GroupProductKeywordsEntity>

    fun findKeywordsForGroupId(groupId: Long?): List<Array<String>>

    fun findGroupsl(): List<GroupEntity>
}
